src/main.cpp

```diff
@@ -3365,27 +3365,9 @@ bool ProcessNewBlock(CValidationState& state, CNode* pfrom, const CBlock* pblock
         // Store to disk
         CBlockIndex* pindex = nullptr;
         bool ret = AcceptBlock(*pblock, state, &pindex, dbp, checked);
+        if (fAccepted) *fAccepted = ret;
```

Comment:
Would be better if we use the return boolean (or return an enum) instead of adding a new argument to the function.

Reply:
No, can't use the return value of `ProcessNewBlock`, because it might fail in `ActivateBestChain`, while we only want to know if `AcceptBlock` passes or not for the spam filter.
We could use an enum as return value,  but that would be only cosmetic and would require a more invasive refactoring. We can do that in a successive PR if it's really worth.

Reply:
sounds good.